File: app/src/main/java/com/remember/app/ui/base/BaseViewHolder.java
package com.remember.app.ui.base;

import androidx.recyclerview.widget.RecyclerView;
import android.view.View;

public abstract class BaseViewHolder extends RecyclerView.ViewHolder {

    private int currentPosition;

    public BaseViewHolder(View itemView) {
        super(itemView);
    }

    public void onBind(int position) {
        currentPosition = position;
    }

    public int getCurrentPosition() {
        return currentPosition;
    }
}

File: app/src/main/java/com/remember/app/ui/base/BaseViewHolder.kt
package com.remember.app.ui.base

import android.view.View
import androidx.recyclerview.widget.RecyclerView

abstract class BaseViewHolder(
    itemView: View,
    private var currentPosition: Int
) : RecyclerView.ViewHolder(itemView) {

    open fun onBind(position: Int) {
        currentPosition = position
    }
}
